lib/util/git/set-branch-commit.ts

```diff
@@ -0,0 +1,33 @@
+import { getCache } from '../cache/repository';
+import type { BranchCache } from '../cache/repository/types';
+import { getBranchCommit } from '.';
+
+/**
+ * Called when a new commit is added to branch
+ *
+ * ie. when branch is created/updated
+ */
+export function setBranchCommit(
```

Comment:
```suggestion
export function setBranchNewCommit(
```

Let's make it as explicit as possible for future maintainability